app/services/catalog/create_approval_request.rb

```diff
@@ -0,0 +1,66 @@
+require 'approval_api_client'
+
+module Catalog
+  class CreateApprovalRequest
+    attr_reader :order
+
+    def initialize(id)
+      @order = Order.find_by!(:id => id)
+
+      ApprovalApiClient.configure do |config|
+        config.api_key['x-rh-auth-identity'] = ManageIQ::API::Common::Request.current.request_id
+      end
+
+      # Set up the instance class-wide since there are N portfolio_items and workflows to request
+      @api_instance = ApprovalApiClient::RequestApi.new
+    end
+
+    def process
+      @order.order_items.each do |item|
+        approvals = send_approval_requests(item)
+
+        approvals.compact.each do |req|
+          @order.approval_requests << create_approval_request(req)
+        end
+      end
+
+      @order.update(:status => "Ordered", :ordered_at => Time.now.utc)
+
+      self
+    rescue ApprovalAPIClient::ApiError => e
+      Rails.logger.error("Error putting in approval Request for #{order.id}: #{e.message}")
+      raise
+    end
+
+    private
+
+    def send_approval_requests(item)
+      [].tap do |requests|
+        workflows(item.portfolio_item).each do |workflow|
+          requests << @api_instance.create_request(workflow, request_body_from(item))
+        end
+      end
+    end
+
+    def request_body_from(item)
+      ApprovalApiClient::RequestIn.new.tap do |request|
+        request.requester = ManageIQ::API::Common::Request.current.user.username
+        request.name      = item.portfolio_item.name
+        request.content   = item.service_parameters.to_json
```

Comment:
Based on the document in https://docs.google.com/document/d/1MJaGiA7TWxz3YgTGyoWjq88DXzeRKHUvjb_GVdTmprI/edit?usp=sharing,
we'd better to have the content structured as
```
{
  'product' => <product>,
  'portfolio' => <portfolio>,
  'platform' => <platform>,
  'order_id' => <orderid>,
  'params' => item.service_parameters.to_json
}
```

Reply:
Updated to construct it this way, without `platform` currently as we don't store that anywhere (yet).